test/spoom/model/model_test.rb

```diff
@@ -0,0 +1,119 @@
+# typed: true
+# frozen_string_literal: true
+
+require "test_helper"
+
+module Spoom
+  class Model
+    class ModelTest < Minitest::Test
+      extend T::Sig
+
+      def test_resolve_symbol
+        model = model(<<~RB)
+          module Foo
+            class Bar
+              class Baz; end
+            end
+          end
+        RB
+
+        context = model["Foo"]
+        assert_equal(model["Foo"], model.resolve_symbol("::Foo", context: context))
+        assert_equal(model["Foo"], model.resolve_symbol("Foo", context: context))
+        assert_equal(model["Foo::Bar"], model.resolve_symbol("Bar", context: context))
+        assert_equal(model["Foo::Bar"], model.resolve_symbol("Foo::Bar", context: context))
+        assert_equal(model["Foo::Bar::Baz"], model.resolve_symbol("Bar::Baz", context: context))
+        assert_equal(model["Foo::Bar::Baz"], model.resolve_symbol("Foo::Bar::Baz", context: context))
+
+        context = model["Foo::Bar"]
+        assert_equal(model["Foo"], model.resolve_symbol("::Foo", context: context))
+        assert_equal(model["Foo"], model.resolve_symbol("Foo", context: context))
+        assert_equal(model["Foo::Bar"], model.resolve_symbol("Bar", context: context))
+        assert_equal(model["Foo::Bar"], model.resolve_symbol("Foo::Bar", context: context))
+        assert_equal(model["Foo::Bar::Baz"], model.resolve_symbol("Baz", context: context))
+        assert_equal(model["Foo::Bar::Baz"], model.resolve_symbol("Bar::Baz", context: context))
+        assert_equal(model["Foo::Bar::Baz"], model.resolve_symbol("Foo::Bar::Baz", context: context))
+
+        context = model["Foo::Bar::Baz"]
+        assert_equal(model["Foo"], model.resolve_symbol("::Foo", context: context))
+        assert_equal(model["Foo"], model.resolve_symbol("Foo", context: context))
+        assert_equal(model["Foo::Bar"], model.resolve_symbol("Bar", context: context))
+        assert_equal(model["Foo::Bar"], model.resolve_symbol("Foo::Bar", context: context))
+        assert_equal(model["Foo::Bar::Baz"], model.resolve_symbol("Baz", context: context))
+        assert_equal(model["Foo::Bar::Baz"], model.resolve_symbol("Bar::Baz", context: context))
+        assert_equal(model["Foo::Bar::Baz"], model.resolve_symbol("Foo::Bar::Baz", context: context))
+
+        context = model["Foo"]
+        assert_instance_of(UnresolvedSymbol, model.resolve_symbol("Qux", context: context))
+        assert_instance_of(UnresolvedSymbol, model.resolve_symbol("::Bar", context: context))
+        assert_instance_of(UnresolvedSymbol, model.resolve_symbol("::Baz", context: context))
+      end
+
+      def test_symbols_hierarchy_for_classes
+        model = model(<<~RB)
+          class A; end
+          class B < A; end
+          class C < B; end
+          class D; end
+        RB
+
+        assert_equal(
+          ["B", "C"],
+          model.subtypes(model["A"]).map(&:full_name).sort,
```

Comment:
Should we not test the order? Why do we sort?

Reply:
Good question: The order is not relevant at this point since we do not do any kind of linearization (yet).